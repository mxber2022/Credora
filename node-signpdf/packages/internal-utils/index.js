var readTestResource = require('./readTestResource');
var createPdfkitDocument = require('./createPdfkitDocument');

module.exports = {
    readTestResource: readTestResource,  
    createPdfkitDocument: createPdfkitDocument,  
}