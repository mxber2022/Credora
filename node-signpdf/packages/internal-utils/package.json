{
  "private": true,
  "name": "@signpdf/internal-utils",
  "version": "3.2.0",
  "description": "",
  "main": "index.js",
  "repository": {
    "type": "git",
    "url": "git+https://github.com/vbuch/node-signpdf.git"
  }
}
