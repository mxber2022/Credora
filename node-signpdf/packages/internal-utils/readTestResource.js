var fs = require('fs');

module.exports = function (filename) {
    if (filename === '.' || (/(\/|\\|\.\.)/).test(filename)) {
        throw new Error(`"${filename}" is invalid filename.`);
    }
    return fs.readFileSync(`${__dirname}/../../resources/${filename}`);
};