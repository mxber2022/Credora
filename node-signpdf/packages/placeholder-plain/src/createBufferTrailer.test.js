import {readTestResource} from '@signpdf/internal-utils';
import createBufferTrailer from './createBufferTrailer';

describe(createBufferTrailer, () => {
    it('creates expected outputs', () => {
        const pdfBuffer = readTestResource('signed.pdf');

        /** @type {import("./readPdf").ReadPdfReturnType} */
        const info = {
            xref: {
                startingIndex: 0,
                maxIndex: 2,
                offsets: new Map([
                    [1, 4163],
                    [2, 4098],
                ]),
            },
            rootRef: '1 0 R',
            root: '\n/Type /Catalog\n/Pages 1 0 R\n/AcroForm 9 0 R\n',
            infoRef: '2 0 R',
            trailerStart: 4510,
            previousXrefs: [],
            xRefPosition: 4220,
        };
        const addedReferences = new Map();

        expect(createBufferTrailer(pdfBuffer, info, addedReferences).toString()).toMatchSnapshot();

        info.infoRef = undefined;
        expect(createBufferTrailer(pdfBuffer, info, addedReferences).toString()).toMatchSnapshot();

        addedReferences.set(3, 4077);
        addedReferences.set(4, 3812);
        expect(createBufferTrailer(pdfBuffer, info, addedReferences).toString()).toMatchSnapshot();
    });
});
