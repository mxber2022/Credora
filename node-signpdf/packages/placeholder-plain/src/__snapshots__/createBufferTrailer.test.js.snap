// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createBufferTrailer creates expected outputs 1`] = `
"xref
0 1
0000000000 65535 f 
trailer
<<
/Size 3
/Root 1 0 R
/Info 2 0 R
/Prev 4220
>>
startxref
4654
%%EOF"
`;

exports[`createBufferTrailer creates expected outputs 2`] = `
"xref
0 1
0000000000 65535 f 
trailer
<<
/Size 3
/Root 1 0 R
/Prev 4220
>>
startxref
4654
%%EOF"
`;

exports[`createBufferTrailer creates expected outputs 3`] = `
"xref
0 1
0000000000 65535 f 
3 1
0000004077 00000 n 
4 1
0000003812 00000 n 
trailer
<<
/Size 3
/Root 1 0 R
/Prev 4220
>>
startxref
4654
%%EOF"
`;
